delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/organizations/OrganizationDeleterSpec.scala

```diff
@@ -0,0 +1,115 @@
+
+  override def munitFixtures: Seq[AnyFixture[_]] = List(projectFixture)
+
+  private lazy val (xas, projects) = projectFixture()
+  private lazy val orgDeleter      = OrganizationDeleter(xas)
+  private val projRef              = ProjectRef.unsafe("org", "myproj")
+  private val fields               = ProjectFields(None, ApiMappings.empty, None, None)
+  private lazy val orgs            = GlobalEventLog(Organizations.definition, logConfig, xas)
+
+  implicit val subject: Subject = Identity.User("Bob", Label.unsafe("realm"))
+  implicit val uuidF: UUIDF     = UUIDF.fixed(orgUuid)
+
+  test("Fail when trying to delete a non-empty organization") {
+    val arrange: IO[Any, Unit] = for {
+      _ <- truncateTables
+      _ <- orgs.evaluate(org, OrganizationCommand.CreateOrganization(org, None, Identity.Anonymous))
+      _ <- projects.create(projRef, fields)
+    } yield ()
+
+    val act: UIO[Either[OrganizationNonEmpty, Unit]] =
+      orgDeleter.delete(org).attempt
+
+    def assert(result: Either[OrganizationNonEmpty, Unit]): IO[Any, Unit] = for {
+      eventPartitions <- queryPartitions("scoped_events")
+      statePartitions <- queryPartitions("scoped_states")
+      fetchedProject  <- projects.fetch(projRef)
+      globalOrgEvent  <- orgs.currentEvents(Offset.Start).compile.to(List).map(_.map(_.value.uuid))
+      globalOrgState  <- orgs.currentStates(Offset.Start).compile.to(List).map(_.map(_.value.uuid))
+    } yield {
+      assertEquals(result, Left(OrganizationNonEmpty(org)))
+      val orgPartition = MD5.hash(org.value)
+      assertEquals(eventPartitions.headOption, Some(s"scoped_events_$orgPartition"))
+      assertEquals(statePartitions.headOption, Some(s"scoped_states_$orgPartition"))
+      assertEquals(fetchedProject.value.ref, projRef)
+      assertEquals(globalOrgState, List(orgUuid))
+      assertEquals(globalOrgEvent, List(orgUuid))
+    }
+
+    arrange >> act >>= assert
+  }
+
+  test("Successfully delete an empty organization") {
+    val arrange: IO[Any, Unit] = for {
+      _ <- truncateTables
+      _ <- orgs.evaluate(org, OrganizationCommand.CreateOrganization(org, None, Identity.Anonymous))
+    } yield ()
+
+    val act: UIO[Either[OrganizationNonEmpty, Unit]] =
+      orgDeleter.delete(org).attempt
+
+    def assert(result: Either[OrganizationNonEmpty, Unit]): IO[Any, Unit] = for {
+      globalEventsDeleted   <- orgs.currentEvents(Offset.Start).compile.to(List).map(_.isEmpty)
+      globalStateDeleted    <- orgs.currentStates(Offset.Start).compile.to(List).map(_.isEmpty)
+      eventPartitionDeleted <- queryPartitions("scoped_events").map(_.isEmpty)
+      statePartitionDeleted <- queryPartitions("scoped_states").map(_.isEmpty)
+    } yield {
+      assertEquals(result, Right(()))
+      assertEquals(eventPartitionDeleted, true)
+      assertEquals(statePartitionDeleted, true)
+      assertEquals(globalStateDeleted, true)
+      assertEquals(globalEventsDeleted, true)
+    }
+
+    arrange >> act >>= assert
```

Comment:
Maybe it's possible to rewrite this to be a little more descriptive? I think this test is saying

Given there is an organisation
When I delete that organisation
All data from that organisation is deleted

But's a little abstracted